### PR TITLE
[MWRAPPER-85] mention distributionType

### PR DESCRIPTION
**Pull request by bmarwell**
This will actually tell the user how to obtain the file `MavenWrapperDownloader.java` which is mentioned in the docs.
Unpopular opinion: I liked the old behaviour better where you could download both the `.jar` and `.java` file and just exclude the .jar file from git if you wanted to. This is easier to understand for users as they now will have to read the docs to find out how to obtain the `.java` file as a last-resort fallback.

----

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MWRAPPER-85) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[MWRAPPER-85] - Fixes bug in ApproximateQuantiles`,
       where you replace `MWRAPPER-85` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [X] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
What about:
https://github.com/apache/maven-wrapper/blob/master/maven-wrapper-plugin/src/site/markdown/usage.md

There is descriptions about `Distribution Types`

**Review by bdemers — APPROVED** [review]

File: src/site/markdown/index.md [hunk 1]
```diff
@@ -94,14 +94,25 @@ binary file `.mvn/wrapper/maven-wrapper.jar`. It is used to bootstrap the
...
+```shell
+# defaults to bin
+mvn wrapper:wrapper -Dtype=source 
+```
+
+You can also chose to opt out of all additional resources except the wrapper scripts:
+
+```shell
+mvn wrapper:wrapper -Dtype=script
+```
```
Comment (slawekjaranowski) [inline src/site/markdown/index.md:109]: Should we mention  new `only-script` type here?

**Comment by jira-importer** [comment]
Resolve #263

**Comment by jira-importer** [comment]
Resolve #263

**Pull request merged**